### ISSUE TITLE
Improve reconciliation cycle observability

### ISSUE THREAD

**Issue by mnencia** [cloudnative-pg/cloudnative-pg#1474]
Currently, the user has no easy way to see if the operator is failing to reconcile the Cluster resource. To improve the observability, I propose to perform the following changes to the Cluster reconciliation cycle:

1) Add a `Synced` condition that is
  - True with reason `ReconcileSuccess`: The most recent reconciliation cycle succeeded
  - False with reason `ReconcileError`: An error occurred while reconciling the resource
    during the last reconciliation cycle.
  - False with reason `ReconcilePaused`: Indicates reconciliation on the managed resource
    is paused via the pause annotation.

2) The operator should Emit an Event every time the reconciliation cycle terminates
   with an error that is not retriable.

Similarly, I would like to have a `PrimarySynced` condition signaling the status of the instance manager reconciliation cycle on the primary (Paused if it is a designated primary) and have the instance manager reconciliation cycle emitting an Event in case of a not retriable error.

### Prior art

Crossplane operator do something similar. See https://github.com/crossplane/crossplane-runtime/blob/master/apis/common/v1/condition.go and, for example https://github.com/crossplane/crossplane/pull/2968/files

**Comment by phisco** [comment]
I think this would be really helpful. We should make sure to add a column to the `kubectl get cluster` output for it. And it could probably be worth adding it to Poolers and the rest of the resources too.

**Comment by sxd** [comment]
Yeah and something we should also add to the metrics we export and the plugin
